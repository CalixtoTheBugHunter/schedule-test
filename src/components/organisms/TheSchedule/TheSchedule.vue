<template>
    <div class="TheSchedule">
        <h3>May</h3>
        <div class="TheSchedule__week">
            <div class="TheSchedule__week-day" v-for="day in weeksDays" :key="day.id">
                {{ day.name }}
            </div>
        </div>
        <div class="TheSchedule__calendar">
            <span 
                class="TheSchedule__calendar-span" 
                v-if="showCalendarSpan" 
                :style="setCalendarSpanSize" />
            <div class="TheSchedule__calendar-day" v-for="(day, i) in months[4].days" :key="`dayOfMonth${i}`">
                {{ day }}
                <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder v-for="reminder in reminders(day)" :key="reminder.id" :title="reminder.title"/>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex'
const { mapGetters } = createNamespacedHelpers('schedule')

import weeksDays from './utils/weeks'
import months from './utils/months'

import { BaseReminder } from '@/components/atoms'

export default {
    name: 'TheSchedule',
    components: { BaseReminder },
    data() {
        return {
            weeksDays,
            months,
            baseDate: '2021-05-01' // Make dynamic this field
        }
    },
    computed: {
        ...mapGetters(['checkDateInData', 'getRemindersByDay']),
        setFirstDateWeekDay() {
            return new Date(this.baseDate).getDay()
        },
        showCalendarSpan() {
            return this.setFirstDateWeekDay < 6
        },
        setCalendarSpanSize() {
            return `width: calc(100px * ${this.setFirstDateWeekDay + 1} + ${this.setFirstDateWeekDay}px)`
        }
    },
    methods: {
        checkDateInDataNonEmpty(day) {
            return this.checkDateInData({ month: 4, date: day}) > 0
        },
        reminders(day) {
            return this.getRemindersByDay({ month: 4, date: day}).reminders
        }
    }
}
</script>

<style lang="scss" scoped>
.TheSchedule {
    color: $color-primary;
    &__week {
        background-color: $color-light;
        display: flex;
        gap: 1px;
        width: 708px;
        padding: 1px;
        box-sizing: border-box;
        &-day {
            flex: 1
        }
    }
    &__calendar {
        background-color: $color-light;
        display: flex;
        gap: 1px;
        flex-flow: wrap;
        width: 708px;
        padding: 1px;
        box-sizing: border-box;

        &-span {
            height: 100px;
        }

        &-day {
            width: 100px;
            height: 100px;
            background: $color-white;
        }
    }
}
</style>