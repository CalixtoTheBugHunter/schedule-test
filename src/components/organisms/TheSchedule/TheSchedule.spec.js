/* eslint-disable no-undef */
import { mount } from '@vue/test-utils'
import TheSchedule from './TheSchedule'
import { createStore } from 'vuex'

describe('TheSchedule', () => {
  let wrapper, store

  beforeEach(() => {

    store = createStore({
      modules: {
        schedule: {
          namespaced: true,
          state() {
            return {
              data: {}
            }
          },
          getters: {
            hasDateInData: () => false,
            getRemindersByDay: () => null
          }
        }
      }
    })

    wrapper = mount(TheSchedule, {
      global: {
        plugins: [store]
      }
    })
  })

  it('Component exists', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('Component not changed', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})