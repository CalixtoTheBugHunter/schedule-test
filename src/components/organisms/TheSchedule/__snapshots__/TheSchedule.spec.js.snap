// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSchedule Component not changed 1`] = `
<div class="TheSchedule">
  <h3>May</h3>
  <div class="TheSchedule__week">
    <div class="TheSchedule__week-day">Sunday</div>
    <div class="TheSchedule__week-day">Monday</div>
    <div class="TheSchedule__week-day">Tuesday</div>
    <div class="TheSchedule__week-day">Wednesday</div>
    <div class="TheSchedule__week-day">Thursday</div>
    <div class="TheSchedule__week-day">Friday</div>
    <div class="TheSchedule__week-day">Saturday</div>
  </div>
  <div class="TheSchedule__calendar"><span class="TheSchedule__calendar-span" style="width: calc(100px * 6 + 5px);"></span>
    <div class="TheSchedule__calendar-day">1
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">2
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">3
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">4
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">5
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">6
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">7
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">8
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">9
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">10
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">11
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">12
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">13
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">14
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">15
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">16
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">17
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">18
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">19
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">20
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">21
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">22
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">23
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">24
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">25
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">26
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">27
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">28
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">29
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">30
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
    <div class="TheSchedule__calendar-day">31
      <!-- <div v-if="checkDateInDataNonEmpty(day)">
                    <base-reminder 
                        v-for="reminder in reminders(day)" 
                        :key="reminder.id" 
                        :title="reminder.title"
                        :color="reminder.color"
                    />
                </div> -->
    </div>
  </div>
</div>
`;
